test/api/importcsv/api_test.go

```diff
@@ -0,0 +1,171 @@
+package importcsv
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+	"github.com/konveyor/tackle2-hub/api"
+	"github.com/konveyor/tackle2-hub/binding"
+	"github.com/konveyor/tackle2-hub/test/assert"
+)
+
+func TestImportCSV(t *testing.T) {
+	for _, r := range TestCases {
+		t.Run(r.FileName, func(t *testing.T) {
+
+			// Upload CSV.
+			inputData := api.ImportSummary{}
+			assert.Must(t, Client.FilePost(api.UploadRoot, r.FileName, &inputData))
+
+			// Inject import summary id into Summary root
+			pathForImportSummary := binding.Path(api.SummaryRoot).Inject(binding.Params{api.ID: inputData.ID})
+
+			// Since uploading the CSV happens asynchronously we need to wait for the upload to check Applications and Dependencies.
+			time.Sleep(time.Second)
+
+			var outputImportSummaries []api.ImportSummary
+			outputMatchingSummary := api.ImportSummary{}
+			for{
+				assert.Should(t, Client.Get(api.SummariesRoot, &outputImportSummaries))
+				for _, gotImport := range outputImportSummaries {
+					if uint(gotImport.ID) == inputData.ID {
+						outputMatchingSummary = gotImport
+					}
+				}
+				if(outputMatchingSummary.ValidCount + outputMatchingSummary.InvalidCount == len(r.ExpectedApplications)+len(r.ExpectedDependencies)){
+					break
+				}
+				time.Sleep(time.Second)
+			}
+
+			// Check list of Applications.
+			gotApps, _ := Application.List()
+			if len(gotApps) != len(r.ExpectedApplications) {
+				t.Errorf("Mismatch in number of imported Applications: Expected %d, Actual %d", len(r.ExpectedApplications), len(gotApps))
+			} else {
+				for i, importedApp := range gotApps {
```

Comment:
This is a tiny thing, not requiring change, but there as a ```gotApps``` array - looks OK and in the loop is ```importedApp```, works, sure, but wouldn't ````gotApp`` match better to the ```gotApps```?

Reply:
Will do the change